### ISSUE TITLE
hyperboloid.py is missing.

### ISSUE THREAD

**Issue by ukaznil** [HazyResearch/hgcn#4]
Hi.
I appreciate this wonderful repository.

When I clone this and tried to operate, I found the "hyperboloid.py" is missing in HEAD, since it has been removed at [the previous commit](https://github.com/HazyResearch/hgcn/commit/afd493aa31dbdcbce8937b914d679d4b5a59f87f).

Could you please add the latest hyperboloid manifold into this repo?

**Comment by NarasimhaM3101** [comment]
Any update on hyperboloid.py please? We are very eager to use this library and test ...

**Comment by vascobailao** [comment]
The problem I found about the hyperboloid.py is when you run it with `--dim 16`. For some reason I'm still trying to figure out, the exponential and logarithmic map return a torch.Tensor full of `na` values.

**Comment by vascobailao** [comment]
If you run it with `--dim 32` works fine on my machine.

**Comment by vascobailao** [comment]
The problem is with the `forward` method of `class HyperbolicGraphConvolution(nn.Module)`. For some reason that I'm trying to figure out, when I add the bias term, the resulting tensor of `h`, is full of `nan` values.

`python train.py --task lp --dataset cora --model HyperGCN --lr 0.01 --dim 16 --num-layers 2 --act relu --bias 1 --dropout 0.5 --weight-decay 0.001 --manifold Hyperboloid --log-freq 5`

**Comment by vascobailao** [comment]
It looks like something in this method 

```    
def forward(self, x):
        # dropConnect
        drop_weight = F.dropout(self.weight, self.dropout, training=self.training)
        mv = self.manifold.mobius_matvec(drop_weight, x, self.c)
        return mv
```

is transforming all values to `nan`.

**Comment by vascobailao** [comment]
The function I'm referring to is:

```
def forward(self, input):
        x, adj = input
        h = self.linear.forward(x)
        if self.bias:
           h = self.bias_add.forward(h)
        h = self.agg.forward(h, adj)
        h = self.hyp_act.forward(h)
        output = h, adj
        return output
```

**Comment by vascobailao** [comment]
Apparently, at some point during training the `x` variable is full of `nan` values.

**Comment by vascobailao** [comment]
@ines-chami

**Comment by ukaznil** [comment]
@vascobailao Hi, nice to meet you.
What you pointed out seems different from this issue, which is mentioning just about the absence of `hyperboloid.py` in this repository.

I suggest that you create a new issue, how about?

**Comment by vascobailao** [comment]
I think its very related. The fact that the `hyperboloid.py` file is not available is because its not working.

**Comment by ukaznil** [comment]
Thanks for your reply, I got it!
Sorry for my immature comment.

**Comment by vascobailao** [comment]
I found some problems with specific operations in the transformation from the hyperbolic manifold to the tangent plane. I still have to find some time to go over the differences between the two manifolds.

**Comment by ines-chami** [comment]
We will be posting code for the hyperboloid manifold shortly. You should be able to obtain similar results on the benchmarks with the PoincareBall model.

**Comment by vascobailao** [comment]
@ines-chami can you confirm the problem is in the mobius addition and multiplication? Any news on when the code will be available?

**Comment by ines-chami** [comment]
Added.
